agent/php_nrini.c

```diff
@@ -1730,6 +1730,45 @@ static PHP_INI_MH(nr_unsigned_int_mh) {
   return SUCCESS;
 }
 
+static PHP_INI_MH(nr_log_events_max_samples_stored_mh) {
+  nriniuint_t* p;
+  int val = NR_DEFAULT_LOG_EVENTS_MAX_SAMPLES_STORED;
+
+#ifndef ZTS
+  char* base = (char*)mh_arg2;
+#else
+  char* base = (char*)ts_resource(*((int*)mh_arg2));
+#endif
+
+  p = (nriniuint_t*)(base + (size_t)mh_arg1);
+
+  (void)entry;
+  (void)mh_arg3;
+  NR_UNUSED_TSRMLS;
+
+  /* Anything other than a valid value will result in the
+   * default value of NR_DEFAULT_LOG_EVENTS_MAX_SAMPLES_STORED.
+   */
+
+  p->where = 0;
+
+  if (0 != NEW_VALUE_LEN) {
+    val = (int)strtol(NEW_VALUE, 0, 0);
```

Comment:
Would be good to check strtol() error conditions.

Reply:
I agree, but am conflicted how to go about it- I could handle the error here and leave it at that, or abstract out a nr_ helper function for strtol() functionality + error handling generic use

Reply:
Currently I do not believe such a function exists in the codebase, and strtol() is used without error checking in several places in the code.

Reply:
I have attempted to address this in commit 9a4bd3ffd24564223c3fc44ee417df789fa4a95c. Currently it has only been integrated into the log event samples INI check, but I believe it should be default for all other strtol() calls in this module.

Reply:
Maybe create a ticket for backlog to add strtol wrapper to axiom/util_number_converter so making this improvement is not forgotten?